From adaddc21ee4f1990373145593924fc176a7ad2f0 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Fri, 13 Dec 2013 01:30:22 +0100
Subject: [PATCH 309/311] CLEANUP: acl: remove useless blind copy-paste from
 sample converters

The second error pointer is not needed, we use memprintf() to report
errors, it's a leftover from a blind copy-paste of the original code.
---
 src/acl.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/src/acl.c b/src/acl.c
index 3fadaf1..c4f1934 100644
--- a/src/acl.c
+++ b/src/acl.c
@@ -287,7 +287,6 @@ struct acl_expr *parse_acl_expr(const char **args, char **err, struct arg_list *
 			conv_expr->conv = conv;
 
 			if (arg != endw) {
-				char *err_msg = NULL;
 				int err_arg;
 
 				if (!conv->arg_mask) {
@@ -298,20 +297,18 @@ struct acl_expr *parse_acl_expr(const char **args, char **err, struct arg_list *
 
 				al->kw = smp->fetch->kw;
 				al->conv = conv_expr->conv->kw;
-				if (make_arg_list(endw + 1, arg - endw - 1, conv->arg_mask, &conv_expr->arg_p, &err_msg, NULL, &err_arg, al) < 0) {
+				if (make_arg_list(endw + 1, arg - endw - 1, conv->arg_mask, &conv_expr->arg_p, err, NULL, &err_arg, al) < 0) {
 					memprintf(err, "ACL keyword '%s' : invalid arg %d in conv method '%s' : %s.",
-						  aclkw->kw, err_arg+1, ckw, err_msg);
-					free(err_msg);
+						  aclkw->kw, err_arg+1, ckw, *err);
 					goto out_free_smp;
 				}
 
 				if (!conv_expr->arg_p)
 					conv_expr->arg_p = empty_arg_list;
 
-				if (conv->val_args && !conv->val_args(conv_expr->arg_p, conv, &err_msg)) {
+				if (conv->val_args && !conv->val_args(conv_expr->arg_p, conv, err)) {
 					memprintf(err, "ACL keyword '%s' : invalid args in conv method '%s' : %s.",
-						  aclkw->kw, ckw, err_msg);
-					free(err_msg);
+						  aclkw->kw, ckw, *err);
 					goto out_free_smp;
 				}
 			}
-- 
1.8.1.5

